oe_theme.theme

```diff
@@ -1592,6 +1594,9 @@ function oe_theme_preprocess_page(array &$variables): void {
     'class' => ['ecl-u-pb-xl'],
     'id' => ['main-content'],
   ]);
+  // Put 'data-ecl-has-menu' attribute only if we have a menu inside
+  // the site header.
```

Comment:
After further testing, it turns out we are always adding the 'data-ecl-has-menu' because $variables['page']['navigation'] is never empty, at the very least it contains the cache information.